### ISSUE TITLE
Generated CRDs have status fields in their output set

### ISSUE THREAD

**Issue by maxsmythe** [kubernetes-sigs/controller-tools#456]
Generated CRDs have blank values in their status field. This appears to interfere with ArgoCD.

Given that manifests should not define status, it may be worth explicitly removing status from the output.

Here is some sample output from running controller-gen in the Gatekeeper repository:

``` bash
$ controller-gen crd:trivialVersions=true paths="./apis/..." paths="./pkg/..." output:stdout

---
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  annotations:
    controller-gen.kubebuilder.io/version: v0.3.0
  creationTimestamp: null
  name: constraintpodstatuses.status.gatekeeper.sh
spec:
  group: status.gatekeeper.sh
  names:
    kind: ConstraintPodStatus
    listKind: ConstraintPodStatusList
    plural: constraintpodstatuses
    singular: constraintpodstatus
  scope: Namespaced
  validation:
    openAPIV3Schema:
      description: ConstraintPodStatus is the Schema for the constraintpodstatuses
        API
      properties:
        apiVersion:
          description: 'APIVersion defines the versioned schema of this representation
            of an object. Servers should convert recognized schemas to the latest
            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
          type: string
        kind:
          description: 'Kind is a string value representing the REST resource this
            object represents. Servers may infer this from the endpoint the client
            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
          type: string
        metadata:
          type: object
        status:
          description: ConstraintPodStatusStatus defines the observed state of ConstraintPodStatus
          properties:
            constraintUID:
              description: Storing the constraint UID allows us to detect drift, such
                as when a constraint has been recreated after its CRD was deleted
                out from under it, interrupting the watch
              type: string
            enforced:
              type: boolean
            errors:
              items:
                description: Error represents a single error caught while adding a
                  constraint to OPA
                properties:
                  code:
                    type: string
                  location:
                    type: string
                  message:
                    type: string
                required:
                - code
                - message
                type: object
              type: array
            id:
              type: string
            observedGeneration:
              format: int64
              type: integer
            operations:
              items:
                type: string
              type: array
          type: object
      type: object
  version: v1beta1
  versions:
  - name: v1beta1
    served: true
    storage: true
status:
  acceptedNames:
    kind: ""
    plural: ""
  conditions: []
  storedVersions: []

```

**Comment by alexellis** [comment]
Rather than:

```
status:
  acceptedNames:
    kind: ""
    plural: ""
  conditions: []
  storedVersions: []
```

We get:

```
status:
  acceptedNames:
    kind: ""
    plural: ""
  conditions: null
  storedVersions: null
```

The `null` causes the following error for some users:

```
error: error validating "inlets.inlets.dev_tunnels.yaml": error validating data: ValidationError(CustomResourceDefinition.status): missing required field "storedVersions" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1beta1.CustomResourceDefinitionStatus; if you choose to ignore these errors, turn validation off with --validate=false
```

https://github.com/inlets/inlets-operator/blob/master/artifacts/crds/inlets.inlets.dev_tunnels.yaml

https://github.com/inlets/inlets-operator/blob/master/hack/update-crds.sh

**Comment by knight42** [comment]
Fixed by #433

**Comment by alexellis** [comment]
If that was fixed 2 months ago then why are we having issues now? Can yo advise which specific version of the tool we need to take for the fix?

**Comment by narahari92** [comment]
I see same issue @alexellis mentioned. Had to manually edit the yaml to replace `null` to `[]`.

**Comment by vijtrip2** [comment]
even the latest versions of controller-tools is still producing following section in CRD.
```
status:
  acceptedNames:
    kind: ""
    plural: ""
  conditions: null
  storedVersions: null
```

**Comment by vijtrip2** [comment]
Reverting the commit b9fd30032a18daa07219de8a574c0e33ae6b913d(":sparkles: Remove status from generated CRDs") produces correct crd version.

**Comment by mcristina422** [comment]
No, it seems like my commit is indeed wrong and did not actually accomplish what I thought it did. I think we should revert my previous PR https://github.com/kubernetes-sigs/controller-tools/pull/472

**Comment by misterkramer** [comment]
Hi! Can we remove status field from generation? The CRD shouldn't include the status field as an input (it's an output field). There will be problems with tools like Pulumi https://github.com/pulumi/pulumi-kubernetes/issues/800

**Comment by maxsmythe** [comment]
/remove-lifecycle rotten

Unless this has been fixed, it looks like people may still be interested in seeing this fixed.

**Comment by maxsmythe** [comment]
/remove-lifecycle-stale

still useful

**Comment by morgante** [comment]
Is there a recommended workaround for this?

**Comment by knight42** [comment]
This issue should be resolved by #630.
/close
